backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/purchasecandidate/C_PurchaseCandidate_StepDef.java

```diff
@@ -0,0 +1,176 @@
+			@NonNull final C_Order_StepDefData orderTable,
+			@NonNull final C_OrderLine_StepDefData orderLineTable,
+			@NonNull final C_PurchaseCandidate_StepDefData purchaseCandidateTable)
+	{
+		this.productTable = productTable;
+		this.orderTable = orderTable;
+		this.orderLineTable = orderLineTable;
+		this.purchaseCandidateTable = purchaseCandidateTable;
+	}
+
+	@And("^after not more than (.*)s, C_PurchaseCandidates are found$")
+	public void find_purchase_candidates(final int timeoutSec, @NonNull final DataTable dataTable) throws InterruptedException
+	{
+		final List<Map<String, String>> tableRows = dataTable.asMaps(String.class, String.class);
+		for (final Map<String, String> row : tableRows)
+		{
+			findPurchaseCandidate(timeoutSec, row);
+		}
+	}
+
+	@And("C_PurchaseCandidates are validated")
+	public void validate_purchase_candidates(@NonNull final DataTable dataTable)
+	{
+		final List<Map<String, String>> tableRows = dataTable.asMaps(String.class, String.class);
+		for (final Map<String, String> row : tableRows)
+		{
+			validatePurchaseCandidate(row);
+		}
+	}
+
+	private void validatePurchaseCandidate(@NonNull final Map<String, String> row)
+	{
+		final String purchaseCandidateIdentifier = DataTableUtil.extractStringForColumnName(row, I_C_PurchaseCandidate.COLUMNNAME_C_PurchaseCandidate_ID + ".Identifier");
+
+		final I_C_PurchaseCandidate purchaseCandidateRecord = purchaseCandidateTable.get(purchaseCandidateIdentifier);
+
+		final BigDecimal qtyToPurchase = DataTableUtil.extractBigDecimalOrNullForColumnName(row, "OPT." + I_C_PurchaseCandidate.COLUMNNAME_QtyToPurchase);
+
+		if (qtyToPurchase != null)
+		{
+			assertThat(purchaseCandidateRecord.getQtyToPurchase()).isEqualTo(qtyToPurchase);
+		}
+	}
+
+	private void findPurchaseCandidate(final int timeoutSec, @NonNull final Map<String, String> row) throws InterruptedException
+	{
+		final String productIdentifier = DataTableUtil.extractStringForColumnName(row, COLUMNNAME_M_Product_ID + ".Identifier");
+		final String orderIdentifier = DataTableUtil.extractStringForColumnName(row, COLUMNNAME_C_OrderSO_ID + ".Identifier");
+		final String orderLineIdentifier = DataTableUtil.extractStringForColumnName(row, COLUMNNAME_C_OrderLineSO_ID + ".Identifier");
+
+		final I_C_Order orderRecord = orderTable.get(orderIdentifier);
+		final I_C_OrderLine orderLineRecord = orderLineTable.get(orderLineIdentifier);
+		final I_M_Product productRecord = productTable.get(productIdentifier);
+
+		final Supplier<Boolean> isPurchaseCandidateCreated = () -> {
+
+			final I_C_PurchaseCandidate purchaseCandidateRecord = queryBL
+					.createQueryBuilder(I_C_PurchaseCandidate.class)
+					.addOnlyActiveRecordsFilter()
+					.addEqualsFilter(COLUMNNAME_C_OrderSO_ID, orderRecord.getC_Order_ID())
+					.addEqualsFilter(COLUMNNAME_C_OrderLineSO_ID, orderLineRecord.getC_OrderLine_ID())
+					.addEqualsFilter(COLUMNNAME_M_Product_ID, productRecord.getM_Product_ID())
+					.create()
```

Comment:
i think you can create this query just once and use it here and in line 134.
(sidenote: in master_integration2 we have `StepDefUtil.tryAndWaitForItem`) which directly returns the record in question)